From 3ca0208e8b2071cef2cb952088679019e430626f Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Fri, 29 Jun 2018 15:50:40 -0500
Subject: [PATCH] Add ${MKL_ROOT}/bin directory for DLLs

---
 cmake/MKL.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cmake/MKL.cmake b/cmake/MKL.cmake
index c404c17..01c6134 100644
--- a/cmake/MKL.cmake
+++ b/cmake/MKL.cmake
@@ -140,6 +140,7 @@ function(detect_mkl LIBNAME)
             HINTS
                 ${MKLREDIST}/mkl
                 ${MKLREDIST}/intel64/mkl
+                ${__mklinc_root}/bin
                 ${__mklinc_root}/lib)
         if(NOT MKLDLL)
             return()
@@ -162,7 +163,9 @@ function(detect_mkl LIBNAME)
         if(WIN32)
             find_file(MKLIOMP5DLL
                 NAMES "libiomp5.dll" "libiomp5md.dll"
-                HINTS ${MKLREDIST}/../compiler ${__mklinc_root}/lib)
+                HINTS ${MKLREDIST}/../compiler
+                ${__mklinc_root}/bin
+                ${__mklinc_root}/lib)
             if(NOT MKLIOMP5DLL)
                 return()
             endif()
-- 
2.7.4

