From 49b4778c9e472f5ca53b9089f4b7bb03ccc63cb1 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Tue, 30 Oct 2018 10:22:04 -0500
Subject: [PATCH 1/6] Include stddef in sasl.h for size_t

---
 include/sasl.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/sasl.h b/include/sasl.h
index fef4d51..d1da45a 100755
--- a/include/sasl.h
+++ b/include/sasl.h
@@ -130,6 +130,8 @@
 #define SASL_VERSION_FULL ((SASL_VERSION_MAJOR << 16) |\
       (SASL_VERSION_MINOR << 8) | SASL_VERSION_STEP)
 
+/* For size_t */
+#include <stddef.h>
 #include "prop.h"
 
 /*************
-- 
2.7.4

