diff --git a/build/generator/gen_win_dependencies.py b/build/generator/gen_win_dependencies.py
index eac7e7f..dd8a2e3 100644
--- a/build/generator/gen_win_dependencies.py
+++ b/build/generator/gen_win_dependencies.py
@@ -566,9 +566,11 @@ class GenDependenciesBase(gen_base.GeneratorBase):
 
     # apr-Util 0.9-1.4 compiled expat to 'xml.lib', but apr-util 1.5 switched
     # to the more common 'libexpat.lib'
-    libname = 'libexpat.lib'
-    if not os.path.exists(os.path.join(lib_dir, 'libexpat.lib')):
-      libname = 'xml.lib'
+    libname = 'xml.lib'
+    if os.path.exists(os.path.join(lib_dir, 'libexpat.lib')):
+      libname = 'libexpat.lib'
+    elif os.path.exists(os.path.join(lib_dir, 'expat.lib')):
+      libname = 'expat.lib'
 
     version = (major, minor, patch)
     xml_version = '%d.%d.%d' % version
