From d7007b877b91a7963e02d100457d03acd6e669f2 Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Thu, 22 Feb 2018 18:26:12 -0600
Subject: [PATCH] remove python jar copy

---
 h2o-py/setup.py | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/h2o-py/setup.py b/h2o-py/setup.py
index 548469cc36..4d02fca959 100644
--- a/h2o-py/setup.py
+++ b/h2o-py/setup.py
@@ -2,7 +2,6 @@
 from setuptools import setup, find_packages
 from codecs import open
 import os
-import shutil
 
 here = os.path.abspath(os.path.dirname(__file__))
 
@@ -19,21 +18,6 @@ with open(os.path.join(here, 'h2o/version.txt'), encoding='utf-8') as f:
 packages = find_packages(exclude=["tests*"])
 print("Found packages: %r" % packages)
 
-
-# Copy h2o.jar to the h2o/backend/bin directory
-h2o_jar_src = os.path.join(here, "..", "build", "h2o.jar")
-h2o_jar_dst = os.path.join(here, "h2o", "backend", "bin", "h2o.jar")
-if not os.path.exists(os.path.dirname(h2o_jar_dst)):
-    os.makedirs(os.path.dirname(h2o_jar_dst))
-
-if os.path.exists(h2o_jar_src):
-    shutil.copyfile(h2o_jar_src, h2o_jar_dst)
-elif os.path.exists(h2o_jar_dst):
-    # The h2o.jar already exists in the target directory -- don't do anything (even if it's an old version)
-    pass
-else:
-    raise RuntimeError("Cannot locate %s to bundle with the h2o package (pwd: %s)." % (h2o_jar_src, here))
-
 setup(
     name='h2o',
 
@@ -89,7 +73,6 @@ setup(
     packages=packages,
     package_data={"h2o": [
         "h2o_data/*.*",     # several small datasets used in demos/examples
-        "backend/bin/*.*",  # h2o.jar core Java library
         "version.txt",      # version file
         "buildinfo.txt"     # buildinfo file
     ]},
-- 
2.11.1

