From b16c83e6932c2afe03aa422272ad682ca7055622 Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Thu, 22 Feb 2018 18:55:10 -0600
Subject: [PATCH] hide r and py builds behind env vars

---
 build.gradle    | 18 ++++++++++--------
 settings.gradle | 11 +++++++++--
 2 files changed, 19 insertions(+), 10 deletions(-)

diff --git a/build.gradle b/build.gradle
index d9ce6bd6bd..af3a5f723f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -107,13 +107,15 @@ ext {
       project(':h2o-scala_2.11'),
     ]
 
-    rProjects = [
-      project(':h2o-r')
-    ]
+    if (project.hasProperty("doR"))
+      rProjects = [project(':h2o-r')]
+    else
+      rProjects = []
 
-    pythonProjects = [
-      project(':h2o-py')
-    ]
+    if (project.hasProperty("doPython"))
+      pythonProjects = [project(':h2o-py')]
+    else
+      pythonProjects = []
 
     // The project which need to be run under CI only
     testNeedsCiProject = [
@@ -269,11 +271,11 @@ subprojects {
         }
     }
 
-    if (project in rProjects) {
+    if (project in rProjects && project.hasProperty("doR")) {
         apply from: "$rootDir/gradle/r.gradle"
     }
 
-    if (project in pythonProjects) {
+    if (project in pythonProjects && project.hasProperty("doPython")) {
         apply from: "$rootDir/gradle/r.gradle"   // same plugins
     }
 
diff --git a/settings.gradle b/settings.gradle
index d09f961165..d72188c470 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -4,8 +4,6 @@ include 'h2o-core'
 include 'h2o-algos'
 include 'h2o-web'
 include 'h2o-app'
-include 'h2o-r'
-include 'h2o-py'
 include 'h2o-assemblies:main'
 include 'h2o-assemblies:genmodel'
 include 'h2o-persist-hdfs'
@@ -29,6 +27,15 @@ include 'h2o-ext-krbstandalone'
 include 'h2o-ext-authsupport'
 include 'h2o-ext-jython-cfunc'
 
+
+if ("true".equals(System.getenv("H2O_BUILD_R"))) {
+   include 'h2o-r'
+}
+
+if ("true".equals(System.getenv("H2O_BUILD_PY"))) {
+   include 'h2o-py'
+}
+
 // GRPC support
 if ("true".equals(System.getenv("H2O_BUILD_GRPC"))) {
     include 'h2o-assemblies:py2o'
-- 
2.11.1

