--- work/CMakeLists.txt.orig	2018-10-29 18:12:55.228380600 -0500
+++ work/CMakeLists.txt	2018-10-29 18:13:52.678806300 -0500
@@ -151,6 +151,11 @@
   target_link_libraries(LIB_LIEF_SHARED ws2_32)
 endif()
 
+if(MSVC)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /bigobj")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /bigobj")
+endif(MSVC)
+
 target_compile_definitions(LIB_LIEF_STATIC PRIVATE -DLIEF_STATIC)
 target_compile_definitions(LIB_LIEF_SHARED PRIVATE -DLIEF_EXPORTS)
 if (LIEF_SUPPORT_CXX14)
