From 89c5994faf63f4f312ae8d49c04fe708ac1354ed Mon Sep 17 00:00:00 2001
From: Daniel Holth <dholth@anaconda.com>
Date: Wed, 11 Aug 2021 12:23:12 -0400
Subject: [PATCH 1/2] inherit ENV

---
 SConstruct | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SConstruct b/SConstruct
index 12978a6..29bcb7e 100644
--- a/SConstruct
+++ b/SConstruct
@@ -30,6 +30,7 @@ env = Environment(
     WHEEL_TAG=full_tag,
     MSVC_VERSION=MSVC_VERSION,
     TARGET_ARCH=TARGET_ARCH,
+    ENV=os.environ,
 )
 
 use_py_limited = "abi3" in full_tag
-- 
2.32.0

