From 88a328f325bb707929d2bf46f7c1190cf0243155 Mon Sep 17 00:00:00 2001
From: Daniel Holth <dholth@anaconda.com>
Date: Wed, 11 Aug 2021 12:23:50 -0400
Subject: [PATCH 2/2] define PY_SSIZE_T_CLEAN to avoid warning

---
 SConstruct | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index 29bcb7e..5a447b9 100644
--- a/SConstruct
+++ b/SConstruct
@@ -50,7 +50,7 @@ extension = env.SharedLibrary(
     LIBPREFIX="",
     SHLIBSUFFIX=SHLIBSUFFIX,
     CPPPATH=[CRYPT_BLOWFISH] + env["CPPPATH"],
-    CPPFLAGS=["-D__SKIP_GNU"],
+    CPPFLAGS=["-D__SKIP_GNU", "-DPY_SSIZE_T_CLEAN"],
     parse_flags="-DNO_BF_ASM" + " -DPy_LIMITED_API=0x03030000"
     if use_py_limited
     else "",
-- 
2.32.0

