Index: work/qtcreator.pri
===================================================================
--- work.orig/qtcreator.pri
+++ work/qtcreator.pri
@@ -94,7 +94,7 @@ isEmpty(IDE_BUILD_TREE) {
 }
 
 IDE_APP_PATH = $$IDE_BUILD_TREE/bin
-osx {
+osx_no_thanks {
     IDE_APP_TARGET   = "$$IDE_DISPLAY_NAME"
 
     # check if IDE_BUILD_TREE is actually an existing Qt Creator.app,
@@ -105,8 +105,9 @@ osx {
     # set output path if not set manually
     isEmpty(IDE_OUTPUT_PATH): IDE_OUTPUT_PATH = $$IDE_APP_BUNDLE/Contents
 
+    IDE_PLUGIN_CASED = Plugins
     IDE_LIBRARY_PATH = $$IDE_OUTPUT_PATH/Frameworks
-    IDE_PLUGIN_PATH  = $$IDE_OUTPUT_PATH/PlugIns
+    IDE_PLUGIN_PATH  = $$IDE_OUTPUT_PATH/$$IDE_PLUGIN_CASED
     IDE_LIBEXEC_PATH = $$IDE_OUTPUT_PATH/Resources/libexec
     IDE_DATA_PATH    = $$IDE_OUTPUT_PATH/Resources
     IDE_DOC_PATH     = $$IDE_DATA_PATH/doc
@@ -114,7 +115,7 @@ osx {
     copydata = 1
 
     LINK_LIBRARY_PATH = $$IDE_APP_BUNDLE/Contents/Frameworks
-    LINK_PLUGIN_PATH  = $$IDE_APP_BUNDLE/Contents/PlugIns
+    LINK_PLUGIN_PATH  = $$IDE_APP_BUNDLE/Contents/$$IDE_PLUGIN_CASED
 
     INSTALL_LIBRARY_PATH = $$QTC_PREFIX/$${IDE_APP_TARGET}.app/Contents/Frameworks
     INSTALL_PLUGIN_PATH  = $$QTC_PREFIX/$${IDE_APP_TARGET}.app/Contents/PlugIns
@@ -130,8 +131,9 @@ osx {
     # target output path if not set manually
     isEmpty(IDE_OUTPUT_PATH): IDE_OUTPUT_PATH = $$IDE_BUILD_TREE
 
+    IDE_PLUGIN_CASED = plugins
     IDE_LIBRARY_PATH = $$IDE_OUTPUT_PATH/$$IDE_LIBRARY_BASENAME/qtcreator
-    IDE_PLUGIN_PATH  = $$IDE_LIBRARY_PATH/plugins
+    IDE_PLUGIN_PATH  = $$IDE_LIBRARY_PATH/$$IDE_PLUGIN_CASED
     IDE_DATA_PATH    = $$IDE_OUTPUT_PATH/share/qtcreator
     IDE_DOC_PATH     = $$IDE_OUTPUT_PATH/share/doc/qtcreator
     IDE_BIN_PATH     = $$IDE_OUTPUT_PATH/bin
@@ -142,10 +144,10 @@ osx {
     !isEqual(IDE_SOURCE_TREE, $$IDE_OUTPUT_PATH):copydata = 1
 
     LINK_LIBRARY_PATH = $$IDE_BUILD_TREE/$$IDE_LIBRARY_BASENAME/qtcreator
-    LINK_PLUGIN_PATH  = $$LINK_LIBRARY_PATH/plugins
+    LINK_PLUGIN_PATH  = $$LINK_LIBRARY_PATH/$$IDE_PLUGIN_CASED
 
     INSTALL_LIBRARY_PATH = $$QTC_PREFIX/$$IDE_LIBRARY_BASENAME/qtcreator
-    INSTALL_PLUGIN_PATH  = $$INSTALL_LIBRARY_PATH/plugins
+    INSTALL_PLUGIN_PATH  = $$INSTALL_LIBRARY_PATH/$$IDE_PLUGIN_CASED
     win32: \
         INSTALL_LIBEXEC_PATH = $$QTC_PREFIX/bin
     else: \
@@ -166,6 +168,7 @@ DEFINES += $$shell_quote(RELATIVE_PLUGIN
 DEFINES += $$shell_quote(RELATIVE_LIBEXEC_PATH=\"$$RELATIVE_LIBEXEC_PATH\")
 DEFINES += $$shell_quote(RELATIVE_DATA_PATH=\"$$RELATIVE_DATA_PATH\")
 DEFINES += $$shell_quote(RELATIVE_DOC_PATH=\"$$RELATIVE_DOC_PATH\")
+DEFINES += $$shell_quote(IDE_PLUGIN_CASED=\"$$IDE_PLUGIN_CASED\")
 
 INCLUDEPATH += \
     $$IDE_BUILD_TREE/src \ # for <app/app_version.h> in case of actual build directory
@@ -176,7 +179,7 @@ INCLUDEPATH += \
 win32:exists($$IDE_SOURCE_TREE/lib/qtcreator) {
     # for .lib in case of binary package with dev package
     LIBS *= -L$$IDE_SOURCE_TREE/lib/qtcreator
-    LIBS *= -L$$IDE_SOURCE_TREE/lib/qtcreator/plugins
+    LIBS *= -L$$IDE_SOURCE_TREE/lib/qtcreator/$$IDE_PLUGIN_CASED
 }
 
 QTC_PLUGIN_DIRS_FROM_ENVIRONMENT = $$(QTC_PLUGIN_DIRS)
Index: work/qtcreator.pro
===================================================================
--- work.orig/qtcreator.pro
+++ work/qtcreator.pro
@@ -107,7 +107,7 @@ linux {
     INSTALLS += appstream desktop
 }
 
-macx {
+macx_no_thanks {
     APPBUNDLE = "$$OUT_PWD/bin/$${IDE_APP_TARGET}.app"
     BINDIST_SOURCE.release = "$$OUT_PWD/bin/$${IDE_APP_TARGET}.app"
     BINDIST_SOURCE.debug = "$$OUT_PWD/bin"
@@ -132,6 +132,9 @@ macx {
             && rmdir /s /q binary-artifacts
         QMAKE_EXTRA_TARGETS += deployartifacts
     }
+    macx {
+        CONFIG -= app_bundle
+    }
 }
 
 INSTALLER_ARCHIVE_FROM_ENV = $$(INSTALLER_ARCHIVE)
Index: work/src/app/app.pro
===================================================================
--- work.orig/src/app/app.pro
+++ work/src/app/app.pro
@@ -32,13 +32,15 @@ win32 {
         RC_VERSION_STRING=\"$${QTCREATOR_DISPLAY_VERSION}\" \
         RC_COPYRIGHT=\"$$replace(COPYRIGHT, " ", "\\x20")\"
     RC_FILE = qtcreator.rc
-} else:macx {
+} else:macx_no_thanks {
     LIBS += -framework CoreFoundation
     QMAKE_ASSET_CATALOGS = $$PWD/qtcreator.xcassets
     QMAKE_ASSET_CATALOGS_BUILD_PATH = $$IDE_DATA_PATH
     QMAKE_ASSET_CATALOGS_INSTALL_PATH = $$INSTALL_DATA_PATH
     QMAKE_ASSET_CATALOGS_APP_ICON = qtcreator
+}
 
+macx: {
     infoplist = $$cat($$PWD/app-Info.plist, blob)
     infoplist = $$replace(infoplist, \\$\\{MACOSX_DEPLOYMENT_TARGET\\}, $$QMAKE_MACOSX_DEPLOYMENT_TARGET)
     infoplist = $$replace(infoplist, \\$\\{IDE_COPYRIGHT_YEAR\\}, $$QTCREATOR_COPYRIGHT_YEAR)
@@ -60,5 +62,6 @@ DISTFILES += qtcreator.rc \
 QMAKE_SUBSTITUTES += $$PWD/app_version.h.in
 
 CONFIG += no_batch
+CONFIG -= app_bundle
 
 QMAKE_EXTRA_TARGETS += deployqt # dummy
Index: work/src/app/main.cpp
===================================================================
--- work.orig/src/app/main.cpp
+++ work/src/app/main.cpp
@@ -224,12 +224,12 @@ static inline QStringList getPluginPaths
     //    "$XDG_DATA_HOME/data/QtProject/qtcreator" or "~/.local/share/data/QtProject/qtcreator" on Linux
     //    "~/Library/Application Support/QtProject/Qt Creator" on Mac
     QString pluginPath = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation);
-    if (Utils::HostOsInfo::isAnyUnixHost() && !Utils::HostOsInfo::isMacHost())
+    if (Utils::HostOsInfo::isAnyUnixHost() /* && !Utils::HostOsInfo::isMacHost() */ )
         pluginPath += QLatin1String("/data");
     pluginPath += QLatin1Char('/')
             + QLatin1String(Core::Constants::IDE_SETTINGSVARIANT_STR)
             + QLatin1Char('/');
-    pluginPath += QLatin1String(Utils::HostOsInfo::isMacHost() ?
+    pluginPath += QLatin1String(false && Utils::HostOsInfo::isMacHost() ?
                                     Core::Constants::IDE_DISPLAY_NAME :
                                     Core::Constants::IDE_ID);
     pluginPath += QLatin1String("/plugins/");
Index: work/src/plugins/qmldesigner/qmldesignerplugin.cpp
===================================================================
--- work.orig/src/plugins/qmldesigner/qmldesignerplugin.cpp
+++ work/src/plugins/qmldesigner/qmldesignerplugin.cpp
@@ -222,10 +222,10 @@ bool QmlDesignerPlugin::initialize(const
 bool QmlDesignerPlugin::delayedInitialize()
 {
     // adding default path to item library plugins
-    const QString pluginPath = Utils::HostOsInfo::isMacHost()
-            ? QString(QCoreApplication::applicationDirPath() + "/../PlugIns/QmlDesigner")
+    const QString pluginPath = (false && Utils::HostOsInfo::isMacHost())
+            ? QString(QCoreApplication::applicationDirPath() + "/../" IDE_PLUGIN_CASED "/QmlDesigner")
             : QString(QCoreApplication::applicationDirPath() + "/../"
-                      + QLatin1String(IDE_LIBRARY_BASENAME) + "/qtcreator/plugins/qmldesigner");
+                      + QLatin1String(IDE_LIBRARY_BASENAME) + "/qtcreator/" IDE_PLUGIN_CASED "/qmldesigner");
     MetaInfo::setPluginPaths(QStringList(pluginPath));
 
     d->settings.fromSettings(Core::ICore::settings());
Index: work/src/qtcreatorlibrary.pri
===================================================================
--- work.orig/src/qtcreatorlibrary.pri
+++ work/src/qtcreatorlibrary.pri
@@ -12,10 +12,14 @@ win32 {
 
 DESTDIR = $$IDE_LIBRARY_PATH
 
-osx {
+osx_no_thanks {
     QMAKE_LFLAGS_SONAME = -Wl,-install_name,@rpath/
     QMAKE_LFLAGS += -compatibility_version $$QTCREATOR_COMPAT_VERSION
 }
+osx {
+    QMAKE_LFLAGS_SONAME = -Wl,-install_name,@rpath/qtcreator/
+    QMAKE_LFLAGS += -compatibility_version $$QTCREATOR_COMPAT_VERSION
+}
 
 RPATH_BASE = $$IDE_LIBRARY_PATH
 include(rpath.pri)
Index: work/src/qtcreatorplugin.pri
===================================================================
--- work.orig/src/qtcreatorplugin.pri
+++ work/src/qtcreatorplugin.pri
@@ -63,7 +63,7 @@ isEmpty(USE_USER_DESTDIR) {
         isEmpty(DESTDIRBASE):DESTDIRBASE = "$$(HOME)/.local/share/data"
         else:DESTDIRBASE = "$$DESTDIRBASE/data"
     }
-    DESTDIR = "$$DESTDIRBASE/QtProject/$$DESTDIRAPPNAME/plugins/$$QTCREATOR_VERSION"
+    DESTDIR = "$$DESTDIRBASE/QtProject/$$DESTDIRAPPNAME/$$IDE_PLUGIN_CASED/$$QTCREATOR_VERSION"
 }
 LIBS += -L$$DESTDIR
 INCLUDEPATH += $$OUT_PWD
@@ -85,7 +85,7 @@ exists($$PLUGINJSON_IN) {
 }
 
 osx {
-    QMAKE_LFLAGS_SONAME = -Wl,-install_name,@rpath/
+    QMAKE_LFLAGS_SONAME = -install_name,@rpath/$$IDE_PLUGIN_CASED/
     QMAKE_LFLAGS += -compatibility_version $$QTCREATOR_COMPAT_VERSION
 }
 
Index: work/src/tools/qtcreatorwidgets/qtcreatorwidgets.pro
===================================================================
--- work.orig/src/tools/qtcreatorwidgets/qtcreatorwidgets.pro
+++ work/src/tools/qtcreatorwidgets/qtcreatorwidgets.pro
@@ -27,7 +27,7 @@ linux-*{  
 
 INCLUDEPATH += ../../../src/libs
 macx {
-    LIBS += -L"../../../bin/Qt Creator.app/Contents/PlugIns"
+    LIBS += -L"../../../bin/Qt Creator.app/Contents/$$IDE_PLUGIN_CASED"
     CONFIG(debug, debug|release):LIBS += -lUtils_debug
     else:LIBS += -lUtils
 } else:win32 {
