diff -urN work.old/src/hostapi/coreaudio/pa_mac_core.c work/src/hostapi/coreaudio/pa_mac_core.c
--- work.old/src/hostapi/coreaudio/pa_mac_core.c	2017-09-15 08:25:32.000000000 -0700
+++ work/src/hostapi/coreaudio/pa_mac_core.c	2017-09-15 08:26:26.000000000 -0700
@@ -629,7 +629,7 @@
 
     VVDBUG(("InitializeDeviceInfo(): macCoreDeviceId=%ld\n", macCoreDeviceId));
 
-    memset(deviceInfo, 0, sizeof(deviceInfo));
+    memset(deviceInfo, 0, sizeof(*deviceInfo));
 
     deviceInfo->structVersion = 2;
     deviceInfo->hostApi = hostApiIndex;
