From c54382b106c26dbf383e86ae22214418d53085f6 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Wed, 4 Apr 2018 18:58:07 +0000
Subject: [PATCH 8/9] Also build libmxnet_static on Windows

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9a1c12f2..126872bd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -521,8 +521,8 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin" AND USE_MXNET_LIB_NAMING)
   add_library(mxnet SHARED ${SOURCE})
   add_library(mxnet_static STATIC ${SOURCE})
 else()
+  list(APPEND MXNET_INSTALL_TARGETS mxnet_static)
   if(UNIX)
-    list(APPEND MXNET_INSTALL_TARGETS mxnet_static)
     add_library(mxnet_static STATIC ${SOURCE})
     # Need an arbitrary source file to trigger CMake to build the library
     add_library(mxnet SHARED)
@@ -531,6 +531,7 @@ else()
     target_link_libraries(mxnet PRIVATE mxnet_static) # Let cmake understand the dependency
   else()
     add_library(mxnet SHARED ${SOURCE})
+    add_library(mxnet_static STATIC ${SOURCE})
   endif()
 endif()
 
@@ -579,6 +580,7 @@ target_link_libraries(mxnet PUBLIC dmlc)
 
 if(MSVC AND USE_MXNET_LIB_NAMING)
   set_target_properties(mxnet PROPERTIES OUTPUT_NAME "libmxnet")
+  set_target_properties(mxnet_static PROPERTIES OUTPUT_NAME "libmxnet_static")
 endif()
 
 if(USE_PROFILER)
-- 
2.16.1

