From c0a58063d0633d09f630572397fe45aa986b3618 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Wed, 27 Dec 2017 07:04:30 -0600
Subject: [PATCH 1/9] Relax constraints on requests, make graphviz optional

---
 python/setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index cf94adf9..b3d08f82 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -28,7 +28,8 @@ if "--inplace" in sys.argv:
 else:
     from setuptools import setup
     from setuptools.extension import Extension
-    kwargs = {'install_requires': ['numpy<=1.13.3,>=1.8.2', 'requests==2.18.4', 'graphviz==0.8.1'], 'zip_safe': False}
+    kwargs = {'install_requires': ['numpy<=1.13.3,>=1.8.2', 'requests<3,>=2.18.4'], 'zip_safe': False,
+              'extras_require': {'graphviz': ['graphviz']}}
 from setuptools import find_packages
 
 with_cython = False
-- 
2.16.1

