From ee960789991a1d0d0a1247293be77f2b8174d5bc Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sat, 30 Dec 2017 10:31:59 -0800
Subject: [PATCH 7/7] Prepend additional {C,LD}FLAGS instead of appending

Because order matters!
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index cf09c2f..d292ead 100644
--- a/Makefile
+++ b/Makefile
@@ -241,11 +241,11 @@ ifeq ($(USE_THREADED_ENGINE), 1)
 endif
 
 ifneq ($(ADD_CFLAGS), NONE)
-	CFLAGS += $(ADD_CFLAGS)
+	CFLAGS := $(ADD_CFLAGS) $(CFLAGS)
 endif
 
 ifneq ($(ADD_LDFLAGS), NONE)
-	LDFLAGS += $(ADD_LDFLAGS)
+	LDFLAGS := $(ADD_LDFLAGS) $(LDFLAGS)
 endif
 
 ifneq ($(USE_CUDA_PATH), NONE)
-- 
1.9.5 (Apple Git-50.3)

