--- work/src/_portaudiomodule.c.orig	2017-03-18 13:07:34.000000000 -0500
+++ work/src/_portaudiomodule.c	2018-08-03 07:10:55.131107400 -0500
@@ -36,12 +36,14 @@
 #define DEFAULT_FRAMES_PER_BUFFER 1024
 /* #define VERBOSE */
 
+#if !defined(min)
 #define min(a, b)           \
   ({                        \
     __typeof__(a) _a = (a); \
     __typeof__(b) _b = (b); \
     _a < _b ? _a : _b;      \
   })
+#endif
 
 /************************************************************
  *
