From ac07fabdd342cf39cec9c4a0e36ee5ff9adefc45 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Tue, 30 Oct 2018 14:26:59 -0500
Subject: [PATCH 4/5] openssl 1.1.1 support on windows

---
 plugins/NTMakefile | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/plugins/NTMakefile b/plugins/NTMakefile
index bab0352f..c2e7912a 100755
--- a/plugins/NTMakefile
+++ b/plugins/NTMakefile
@@ -64,6 +64,9 @@ compat_objs = getaddrinfo.obj getnameinfo.obj
 common_sources = plugin_common.c plugin_common.h
 common_objs = plugin_common.obj $(compat_objs)
 
+crypto_compat_sources = crypto-compat.c
+crypto_compat_objs = crypto-compat.obj
+
 saslANONYMOUS_sources = anonymous.c anonymous_init.c $(common_sources)
 saslANONYMOUS_objs = anonymous.obj anonymous_init.obj $(common_objs)
 saslANONYMOUS_out = saslANONYMOUS.dll saslANONYMOUS.exp saslANONYMOUS.lib
@@ -96,8 +99,8 @@ saslGSSAPI_sources = gssapi.c gssapiv2_init.c $(common_sources)
 saslGSSAPI_objs = gssapi.obj gssapiv2_init.obj $(common_objs)
 saslGSSAPI_out = saslGSSAPI.dll saslGSSAPI.exp saslGSSAPI.lib
 
-saslSRP_sources = srp.c srp_init.c $(common_sources)
-saslSRP_objs = srp.obj srp_init.obj $(common_objs)
+saslSRP_sources = srp.c srp_init.c $(common_sources) $(crypto_compat_sources)
+saslSRP_objs = srp.obj srp_init.obj $(common_objs) $(crypto_compat_objs)
 saslSRP_out = saslSRP.dll saslSRP.exp saslSRP.lib
 
 saslOTP_sources = otp.c otp_init.c $(common_sources)
@@ -171,7 +174,8 @@ EXTRA_FLAGS = /D TARGET_WIN_SYSTEM=$(TARGET_WIN_SYSTEM) $(EXTRA_FLAGS)
 EXTRA_FLAGS=$(EXTRA_FLAGS) $(DB_FLAGS) $(OPENSSL_FLAGS) $(GSS_FLAGS) $(SRP_FLAGS) $(SQL_FLAGS) $(DIGEST_FLAGS) $(CRAM_FLAGS) $(SCRAM_FLAGS) $(LDAP_FLAGS)
 CPPFLAGS = /I "..\win32\include" /I "." /I "..\include" $(EXTRA_FLAGS) /D "WIN32" /D "_WIN32" /D "_WINDOWS" /D "_MBCS" /D "_USRDLL"
 
-OPENSSL_LIBS=/libpath:$(OPENSSL_LIBPATH) libeay32.lib ssleay32.lib
+DB_LIBS=/libpath:$(DB_LIBPATH) $(DB_LIB)
+OPENSSL_LIBS=/libpath:$(OPENSSL_LIBPATH) libcrypto.lib libssl.lib
 
 # Where to install files from this directory
 libdir = $(prefix)\lib
@@ -212,8 +216,14 @@ allockey.c: ..\sasldb\allockey.c
 db_berkeley.c: ..\sasldb\db_berkeley.c
 	xcopy /D /Y ..\sasldb\db_berkeley.c .
 
+<<<<<<< HEAD
 db_lmdb.c: ..\sasldb\db_lmdb.c
 	xcopy /D /Y ..\sasldb\db_lmdb.c .
+=======
+crypto-compat.c: ..\crypto-compat.c
+	xcopy /D /Y ..\crypto-compat.c .
+	xcopy /D /Y ..\crypto-compat.h .
+>>>>>>> openssl 1.1.1 support on windows
 
 #Add /pdb: option?
 
-- 
2.27.0

