From b23c3e097a73defb13248b4522d07393a9366183 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 24 Sep 2020 12:16:35 +0200
Subject: [PATCH 5/5] win32: Add anonymous.obj to libsasl_objs

---
 lib/NTMakefile | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/lib/NTMakefile b/lib/NTMakefile
index ff982eb1..76e7b169 100755
--- a/lib/NTMakefile
+++ b/lib/NTMakefile
@@ -9,8 +9,8 @@ compat_objs = getaddrinfo.obj getnameinfo.obj
 !ENDIF
 
 
-libsasl_sources = auxprop.c canonusr.c checkpw.c client.c common.c config.c external.c md5.c saslutil.c server.c seterror.c windlopen.c getsubopt.c plugin_common.c plugin_common.h $(compat_sources)
-libsasl_objs = auxprop.obj canonusr.obj checkpw.obj client.obj common.obj config.obj external.obj md5.obj saslutil.obj server.obj seterror.obj windlopen.obj getsubopt.obj plugin_common.obj $(compat_objs)
+libsasl_sources = auxprop.c canonusr.c checkpw.c client.c common.c config.c external.c md5.c saslutil.c server.c seterror.c windlopen.c getsubopt.c plugin_common.c plugin_common.h anonymous.c $(compat_sources)
+libsasl_objs = auxprop.obj canonusr.obj checkpw.obj client.obj common.obj config.obj external.obj md5.obj saslutil.obj server.obj seterror.obj anonymous.obj windlopen.obj getsubopt.obj plugin_common.obj $(compat_objs)
 libsasl_res = libsasl.res
 libsasl_out = libsasl.dll libsasl.exp libsasl.lib $(libsasl_res)
 
@@ -58,6 +58,10 @@ plugin_common.c: ..\common\plugin_common.c plugin_common.h
 plugin_common.h: ..\common\plugin_common.h
 	xcopy /D /Y ..\common\plugin_common.h .
 
+anonymous.c: ..\plugins\anonymous.c
+	xcopy /D /Y ..\plugins\anonymous.c .
+
+
 auxprop.obj checkpw.obj client.obj common.obj external.obj plugin_common.obj server.obj seterror.obj: ..\include\saslplug.h
 
 auxprop.obj canonusr.obj checkpw.obj client.obj common.obj config.obj external.obj getsubopt.obj md5.obj plugin_common.obj server.obj seterror.obj windlopen.obj: ..\include\sasl.h ..\include\prop.h
@@ -72,6 +76,7 @@ CLEAN :
 	-@erase $(all_out)
 	-@erase plugin_common.h
 	-@erase plugin_common.c
+	-@erase anonymous.c
 	-@erase $(exclude_list)
 
 $(libsasl_res): NTMakefile
-- 
2.27.0

