--- torch/csrc/autograd/python_function.cpp
+++ torch/csrc/autograd/python_function.cpp
@@ -1003,16 +1003,20 @@

 PyObject *THPFunction_saved_tensors(THPFunction *self, void *_unused)
 {
+  HANDLE_TH_ERRORS
   return unpack_saved_variables(self, [](const Variable& var) {
     return createPyObject(var.data());
   });
+  END_HANDLE_TH_ERRORS
 }

 PyObject *THPFunction_saved_variables(THPFunction *self, void *_unused)
 {
+  HANDLE_TH_ERRORS
   return unpack_saved_variables(self, [](const Variable& var) {
     return THPVariable_Wrap(var);
   });
+  END_HANDLE_TH_ERRORS
 }

 PyObject *THPFunction_next_functions(THPFunction *self, void *_unused)

