From 832af069c7ccd88b57bff14c86203cd63d43b1d4 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Fri, 25 Jan 2019 17:12:26 -0600
Subject: [PATCH 4/5] pass module linker flags to cmake

---
 tools/build_pytorch_libs.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/build_pytorch_libs.sh b/tools/build_pytorch_libs.sh
index 23dd0c8..882a45f 100755
--- a/tools/build_pytorch_libs.sh
+++ b/tools/build_pytorch_libs.sh
@@ -243,6 +243,7 @@ function build_caffe2() {
 		       -DCMAKE_CXX_FLAGS="$USER_CFLAGS" \
 		       -DCMAKE_EXE_LINKER_FLAGS="$LDFLAGS $USER_LDFLAGS" \
 		       -DCMAKE_SHARED_LINKER_FLAGS="$LDFLAGS $USER_LDFLAGS" \
+		       -DCMAKE_MODULE_LINKER_FLAGS="$LDFLAGS $USER_LDFLAGS" \
 		       $GLOO_FLAGS \
 		       -DTHD_SO_VERSION=1 \
 		       $THD_FLAGS \
-- 
2.7.4

