diff --git a/setup.py b/setup.py
index bfac6914e..09e2fb231 100644
--- a/setup.py
+++ b/setup.py
@@ -567,7 +567,7 @@ C = Extension("torch._C",
               extra_compile_args=main_compile_args + extra_compile_args,
               include_dirs=include_dirs,
               library_dirs=library_dirs,
-              extra_link_args=extra_link_args + main_link_args + [make_relative_rpath('lib')],
+              extra_link_args=extra_link_args + main_link_args + ['-Wl,-z,lazy'] + [make_relative_rpath('lib')],
               )
 extensions.append(C)
 
