diff --git a/third_party/ideep/CMakeLists.txt b/CMakeLists.txt
index 497c231..1f0e707 100644
--- a/third_party/ideep/CMakeLists.txt
+++ b/third_party/ideep/CMakeLists.txt
@@ -29,7 +29,7 @@ install(DIRECTORY include/ DESTINATION include)
 add_subdirectory(tests)
 
 if (multinode)
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -mavx -Wall -Wextra -Werror")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -Wall -Wextra")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-result")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-parameter")
diff --git a/cmake/settings.cmake b/cmake/settings.cmake
index b361dba..e58b4a7 100644
--- a/third_party/ideep/cmake/settings.cmake
+++ b/third_party/ideep/cmake/settings.cmake
@@ -10,7 +10,7 @@ if (MSVC)
   set(__cxx_flags "/std:c++14 /openmp /Qpar /arch:AVX2")
 else ()
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c11")
-  set(__cxx_flags "-std=c++11 -fvisibility-inlines-hidden -Wall -Werror -Wno-sign-compare -Wno-unknown-pragmas -fvisibility-inlines-hidden -march=native -mtune=native -pthread -fopenmp")
+  set(__cxx_flags "-std=c++11 -fvisibility-inlines-hidden -Wall -Wno-sign-compare -Wno-unknown-pragmas -fvisibility-inlines-hidden -pthread -fopenmp")
 endif()
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${__cxx_flags}")
diff --git a/python/setup.py b/python/setup.py
index 5164f1a..4b69910 100644
--- a/third_party/ideep/python/setup.py
+++ b/third_party/ideep/python/setup.py
@@ -116,7 +116,6 @@ if sys.version_info.major < 3:
     swig_opts += ['-DNEWBUFFER_ON']
 
 ccxx_opts = ['-std=c++11', '-Wno-unknown-pragmas',
-             '-march=native', '-mtune=native',
              '-D_TENSOR_MEM_ALIGNMENT_=4096']
 
 env = os.environ
