diff --git a/third_party/mkl/BUILD b/third_party/mkl/BUILD
index b27d341404..07d78f4b06 100644
--- a/third_party/mkl/BUILD
+++ b/third_party/mkl/BUILD
@@ -18,7 +18,7 @@ load(
 cc_library(
     name = "intel_binary_blob",
     srcs = if_mkl([
-        "@mkl//:libmklml_intel.so",
+        "@mkl//:libmkl_rt.so",
         "@mkl//:libiomp5.so",
     ]),
     visibility = ["//visibility:public"],
diff --git a/third_party/mkl/mkl.BUILD b/third_party/mkl/mkl.BUILD
index 8db97232e1..95b5b23532 100644
--- a/third_party/mkl/mkl.BUILD
+++ b/third_party/mkl/mkl.BUILD
@@ -18,8 +18,8 @@ cc_library(
 )
 
 filegroup(
-    name = "libmklml_intel.so",
-    srcs = ["lib/libmklml_intel.so"],
+    name = "libmkl_rt.so",
+    srcs = ["lib/libmkl_rt.so"],
     visibility = ["//visibility:public"],
 )
 
