From 417267c6ab22b57c135212e883c4e24359d51efd Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 19 Jan 2019 01:29:55 -0600
Subject: [PATCH 1/2] Do not link to rt nor pthread unless we are on Linux

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 79b2b53..9cfc121 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -182,9 +182,9 @@ foreach(target ${TARGETS})
 	set_target_properties(${target} PROPERTIES VERSION ${LIB_VERSION_STRING} SOVERSION ${LIB_VERSION_MAJOR})
 	target_include_directories(${target} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
 	target_link_libraries(${target} ${LIBS})
-	if (UNIX)
+	if (LINUX)
 		target_link_libraries(${target} rt pthread)
-	endif (UNIX)
+	endif (LINUX)
 
 	get_target_property(target_type ${target} TYPE)
 	if (target_type STREQUAL "STATIC_LIBRARY")
-- 
2.17.2 (Apple Git-113)

