--- tapi.orig/lib/Core/ArchitectureSupport.cpp	2017-09-07 02:14:25.000000000 -0700
+++ tapi/lib/Core/ArchitectureSupport.cpp	2017-09-07 02:16:53.000000000 -0700
@@ -15,7 +15,7 @@
 #include "llvm/ADT/StringExtras.h"
 #include "llvm/ADT/StringSwitch.h"
 #include "llvm/Support/Format.h"
-#include "llvm/Support/MachO.h"
+#include "llvm/BinaryFormat/MachO.h"
 #include "llvm/Support/raw_ostream.h"
 #include "tapi/Core/ArchitectureSupport.h"
 
